server_test.go

```diff
@@ -694,6 +694,58 @@ func TestServer(t *testing.T) {
 			assert.Equal(t, "cannot start the server, context is canceled", err.Error())
 		}
 	})
+
+	t.Run("StartWithCustomListenConfig", func(t *testing.T) {
```

Comment:
That's too bad there is no way for us to check the `*net.TCPListener.lc` in this test. Currently, we know the server works but the test doesn't ensure the listen config is actually used. We could use reflection and `unsafe.Pointer` but I'd rather not.

Can you add a second test with a `ListenConfig.Control` function that always returns an error, and check for this error in the assertions? This will ensure the given ListenConfig is actually used.

Can you please add an assertion on `server.listenConfig` in the `"New"` test to check the value from the config is copied to the server? (The same way I do with the HTTP2 config)
```go
assert.Same(t, customListenConfig, s.server.listenConfig)
```